### PR TITLE
analysis dashboard access to satellite bucket (#482)

### PR DESCRIPTION
**Pull request by peterdudfield**
* added var s3_nwp_buckets

* analysis dashboard access to satellite bucket

* added variables.tfvars

* Refactor: Modify s3_nwp_bucket variable to accept a list of bucket ARNs

* updated modules in india/dev & nowcasting/dev to include the new list of s3_nwp_buckets

# Pull Request

## Description

_Please delete the italicised instruction text!
Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context. List any dependencies that are required for this change._

Fixes #

## How Has This Been Tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration_

- [ ] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

- [ ] Yes

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by aryanbhosale** [comment]
> * added var s3_nwp_buckets
> * analysis dashboard access to satellite bucket
> * added variables.tfvars
> * Refactor: Modify s3_nwp_bucket variable to accept a list of bucket ARNs
> * updated modules in india/dev & nowcasting/dev to include the new list of s3_nwp_buckets
> 
> # Pull Request
> ## Description
> _Please delete the italicised instruction text! Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change._
> 
> Fixes #
> 
> ## How Has This Been Tested?
> _Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration_
> 
> * [ ]  Yes
> 
> _If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_
> 
> * [ ]  Yes
> 
> ## Checklist:
> * [ ]  My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
> * [ ]  I have performed a self-review of my own code
> * [ ]  I have made corresponding changes to the documentation
> * [ ]  I have added tests that prove my fix is effective or that my feature works
> * [ ]  I have checked my code and corrected any misspellings

hi @peterdudfield , is there anything that needs to be added here? or is it good to be merged?

**Comment by peterdudfield** [comment]
can you resolved conflicts?

**Comment by aryanbhosale** [comment]
> can you resolved conflicts?

where do i check the conflicts?

**Comment by aryanbhosale** [comment]
> > can you resolved conflicts?
> 
> where do i check the conflicts?

should i open a pr from main to this branch and merge it that way?

**Comment by aryanbhosale** [comment]
> > > can you resolved conflicts?
> > 
> > 
> > where do i check the conflicts?
> 
> should i open a pr from main to this branch and merge it that way?

or should i just add the changes ive made in these 5 files to the main one, thatll be faster in my opinion. what do you think @peterdudfield ?

**Comment by peterdudfield** [comment]
> > > > can you resolved conflicts?
> > > 
> > > 
> > > where do i check the conflicts?
> > 
> > 
> > should i open a pr from main to this branch and merge it that way?
> 
> or should i just add the changes ive made in these 5 files to the main one, thatll be faster in my opinion. what do you think @peterdudfield ?

On this PR, I think its good to resolve conflicts, theres a button in the Github that should do this, or you can do it locally.

**Comment by aryanbhosale** [comment]
> > > > > can you resolved conflicts?
> > > > 
> > > > 
> > > > where do i check the conflicts?
> > > 
> > > 
> > > should i open a pr from main to this branch and merge it that way?
> > 
> > 
> > or should i just add the changes ive made in these 5 files to the main one, thatll be faster in my opinion. what do you think @peterdudfield ?
> 
> On this PR, I think its good to resolve conflicts, theres a button in the Github that should do this, or you can do it locally.

I think the button is visible to the creator of the pr, I think we need to create another pr from main(which is ahead by 25/26 commits)to external-pr(this branch) , and there we'll see what's missing/conflicting, should I do that?

File: terraform/nowcasting/development/main.tf [hunk 1]
```diff
@@ -447,6 +450,9 @@ module "pvsite_api" {
   container-registry = "openclimatefix"
   eb-app_name    = "sites-api"
   eb-instance_type = "t3.small"
+  s3_nwp_bucket = [
+    { bucket_read_policy_arn = module.s3.iam-policy-s3-nwp-read.arn }
+  ]
```
Comment (peterdudfield) [inline terraform/nowcasting/development/main.tf:455]: not sure why pvsite_api is changing

**Comment by peterdudfield** [comment]
> > > > > can you resolved conflicts?
> > > > 
> > > > 
> > > > where do i check the conflicts?
> > > 
> > > 
> > > should i open a pr from main to this branch and merge it that way?
> > 
> > 
> > or should i just add the changes ive made in these 5 files to the main one, thatll be faster in my opinion. what do you think @peterdudfield ?
> 
> On this PR, I think its good to resolve conflicts, theres a button in the Github that should do this, or you can do it locally.

Thanks, ive merge this, but I think there is still something missing. 

need to add the satellite read policy to the analysis dashboard?, So adding [this policy](module.s3.iam-policy-s3-sat-read.arn) to [here](https://github.com/openclimatefix/ocf-infrastructure/blob/main/terraform/nowcasting/development/main.tf#L374)

**Comment by aryanbhosale** [comment]
> > > > > > can you resolved conflicts?
> > > > > 
> > > > > 
> > > > > where do i check the conflicts?
> > > > 
> > > > 
> > > > should i open a pr from main to this branch and merge it that way?
> > > 
> > > 
> > > or should i just add the changes ive made in these 5 files to the main one, thatll be faster in my opinion. what do you think @peterdudfield ?
> > 
> > 
> > On this PR, I think its good to resolve conflicts, theres a button in the Github that should do this, or you can do it locally.
> 
> Thanks, ive merge this, but I think there is still something missing.
> 
> need to add the satellite read policy to the analysis dashboard?, So adding [this policy](module.s3.iam-policy-s3-sat-read.arn) to [here](https://github.com/openclimatefix/ocf-infrastructure/blob/main/terraform/nowcasting/development/main.tf#L374)

What do you think could be added?

**Pull request merged**